neasqc_wp61/models/quantum/embedder.py

```diff
@@ -0,0 +1,322 @@
+        """
+        Saves the dataset containing the embeddings as a TSV file.
+
+        Parameters
+        ----------
+        path : str
+            The path to the location where the user wishes to save the
+            generated dataset containing the embeddings (it should not
+            include a / at the end).
+        filename: str
+            The name with which the generated dataset containing the
+            embeddings will be saved (it should not include .csv or any
+            other file extension).
+        """
+        self.dataset.to_csv(path + "/" + filename + ".tsv", index=False, sep="\t")
+
+
+class Bert(Embedder):
+    """
+    Class for generating BERT embeddings
+    """
+
+    def __init__(
+        self,
+        dataset: DataFrame,
+        sentence_or_word: bool = True,
+        cased_or_uncased: bool = False,
+        **kwargs
+    ) -> None:
+        """
+        Initialises the BERT embedder class
+
+        Parameters
+        ----------
+        dataset : pd.DataFrame
+            Pandas dataset. Each row of the dataset correspods to a
+            sentence.
+        sentence_or_word : bool
+            States whether we want to produce a sentence embedding or
+            word embedding vector for each sentence in the dataset. True
+            is for sentence embedding, False is for word embedding.
+        case_or_uncased: bool
+            States whether we want to work with BERT's cased or uncased
+            pretrained base model. True is for cased, False is for
+            uncased.
+        **kwargs
+            Additional arguments to be passed to the BertTokenizer
+            object. These can be found in
+            https://huggingface.co/docs/transformers/main_classes/tokenizer#transformers.PreTrainedTokenizer
+        """
+        super().__init__(dataset, sentence_or_word)
+        self.cased_or_uncased = cased_or_uncased
+        self.kwargs = kwargs
+
+    def compute_embeddings(self, **kwargs) -> DataFrame:
+        """
+        Creates BERT embeddings for sentences in a dataset
+
+        Parameters
+        ----------
+        **kwargs
+            Additional arguments to be passed to the BertTokenizer
+            object. These can be found in
+            https://huggingface.co/docs/transformers/main_classes/tokenizer#transformers.PreTrainedTokenizer
+
+        Returns
+        -------
+        embeddings_df : pd.DataFrame
+            A pandas dataframe with the same structure as the original
+            dataset, but with an additional column containing the
+            BERT embeddings corresponding to the sentence contained in each
+            row.
+        """
+
+        embeddings_df = self.dataset.copy()
+        embeddings_df.columns = ["class", "sentence", "sentence_structure"]
+
+        if self.cased_or_uncased:
+            tokenizer = BertTokenizer.from_pretrained("bert-base-cased", **kwargs)
+            bert_model = BertModel.from_pretrained("bert-base-cased")
```

Comment:
I would probably do an if to create a string containing "case" or uncased and plug that into the larger string that I'll pass to the tokenizer/model.
You must have copy-pasted these lines because everything except case/uncased is the same, in prog this is a good indicator that you should do things differently 😉 .

Reply:
great suggestion Ryk, thanks!

Reply:
addressed